peachjam/adapters/gazettes.py

```diff
@@ -23,15 +23,34 @@ def check_for_updates(self, last_refreshed):
             f"Checking for new gazettes from Gazettes.Africa since {last_refreshed}"
         )
 
-        new_gazettes = (
-            CoreDocument.objects.filter(
-                doc_type="gazette", jurisdiction=self.jurisdiction
-            )
+        queryset = (
+            CoreDocument.objects.filter(doc_type="gazette")
             .non_polymorphic()
             .using("gazettes_africa")
         )
+
+        if "-" not in self.jurisdiction:
+            # locality code not specified hence None
+            queryset = queryset.filter(jurisdiction=self.jurisdiction, locality=None)
+        else:
+            self.jurisdiction.split("-")
+            jurisdiction = self.jurisdiction[0]

```

Comment:
You *always* want to filter on jurisdiction code, you can do that before the ifs. Then the if statement is just about locality.